File: src/test/java/org/neo/gomina/model/scm/impl/DummyScmClientTest.java
package org.neo.gomina.model.scm.impl;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.junit.Test;
import org.neo.gomina.model.scm.Commit;
import org.neo.gomina.model.scm.dummy.DummyScmClient;

import java.util.List;

public class DummyScmClientTest {

    private final static Logger logger = LogManager.getLogger(DummyScmClientTest.class);

    @Test
    public void getLog() throws Exception {
        DummyScmClient client = new DummyScmClient();
        List<Commit> log = client.getLog("OMS/Server/tradex-basketmanager", "0", 100);
        for (Commit commit : log) {
            logger.info(commit);
        }
    }

    @Test
    public void getFile() throws Exception {
        DummyScmClient client = new DummyScmClient();
        String file = client.getFile("OMS/Server/tradex-basketmanager", "-1");
        logger.info(file);
    }

}

File: src/test/java/org/neo/gomina/integration/scm/dummy/DummyScmClientTest.kt
package org.neo.gomina.integration.scm.dummy

import org.apache.logging.log4j.LogManager
import org.junit.Test

class DummyScmClientTest {

    @Test
    @Throws(Exception::class)
    fun getLog() {
        val client = DummyScmClient("OMS/Server/tradex-basketmanager")
        val log = client.getLog("trunk", "0", 100)
        log.forEach { logger.info(it) }
    }

    @Test
    @Throws(Exception::class)
    fun getFile() {
        val client = DummyScmClient("OMS/Server/tradex-basketmanager")
        val file = client.getFile("trunk", "doc.md", "-1")
        logger.info(file)
    }

    companion object {
        private val logger = LogManager.getLogger(DummyScmClientTest::class.java)
    }

}